assembl/tests/views_tests/test_api.py

```diff
@@ -138,6 +138,41 @@ def test_get_ideas(discussion, test_app, test_session, test_webrequest):
     assert len(ideas) == num_ideas + 1
 
 
+def test_edit_idea_title(test_app, test_session, discussion, discussion_admin_user, subidea_1):
+    """As a discussion admin, I can edit the title of an idea (I don't get an Unauthorized error for editing langstring properties of an idea)"""
+
+    # Step 1: Send a PUT API call to modify idea's title, and check that response is 200 and not Unauthorized or 404
+
+    # Method LangString.generic_json() has not been implemented yet. When it will be done, we would be able to just use: 
```

Comment:
On second thought, it's not a bug.
The thing to do is to set
`ls.owner_object = subidea_1`
before the call to `generic_json`

Reply:
Here https://github.com/assembl/assembl/blob/develop/assembl/models/langstrings.py#L584 it looked like it was not implemented. And when I tried ls.generic_json() it did not produce the result I expected (but I don't remember exactly)

Reply:
When I edit the title of an idea from the v1 UI, the payload sent to APIv1 concerning the shortTitle langstring does not mention any owner_object. I wanted to replicate in the test the way our UI uses it. Is it possible to make the generic_json method work without owner_object?

Reply:
The commented implementation was an override for empty strings, but the base implementation works... if the owner_object is specified.
No, it cannot be made to work otherwise, because the failure is due to security reason; we have no information on the langstring's intrinsic security status, which is inherited from the owner_object.

Reply:
P.S. the owner_object is set by the api. Here, you were attempting to create a langstring outside the api, and get its generic_json method; this is not "the way our api uses it", as such. Hope that explains the failure.